### PR TITLE
SignCommand type improvement

### LINKED ISSUE #16: SignCommand types
ℹ️ This issue is part of Epic #10 

## Checklist

- [x] signature
- [x] signatures_list
- [x] signed_signature_with_hash
- [x] unsigned_signature_with_hash
- [x] signature_with_hash
- [x] signed_command

### PR DESCRIPTION
**Pull request by boterop**
## Description

SignCommand type has to check itself if it is SignedSignatureWithHash or UnsignedSignatureWithHash giving just the params [hash, sig, pub_key]

Fixes #16 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: lib/types/sign_command.ex [hunk 1]
```diff
@@ -15,5 +20,25 @@ defmodule Kadena.Types.SignCommand do
 
   @impl true
   def new(%SignatureWithHash{} = signature), do: %__MODULE__{command: signature}
-  def new(_sig), do: {:error, :invalid_sign_command}
+
+  def new(args) when is_list(args) do
+    hash = Keyword.get(args, :hash, nil)
+    sig = Keyword.get(args, :sig, nil)
+    pub_key = Keyword.get(args, :pub_key, nil)
```
Comment (juanhurtado10) [inline lib/types/sign_command.ex]: @santiagoboterokommit it is not necessary to define nil as the default value, the `Keyword.get/3` returns it by default.

File: lib/types/sign_command.ex [hunk 2]
```diff
@@ -15,5 +20,25 @@ defmodule Kadena.Types.SignCommand do
...
+
+    case validate(hash, sig, pub_key) do
+      {:error, _error} -> {:error, :invalid_sign_command}
+      signature -> %__MODULE__{command: SignatureWithHash.new(signature)}
+    end
+  end
+
+  def new(_args), do: {:error, :invalid_sign_command}
+
+  @spec validate(hash :: String.t(), sig :: Signature.t(), pub_key :: String.t()) ::
```
Comment (juanhurtado10) [inline lib/types/sign_command.ex]: @santiagoboterokommit IMO, **validate** is not an appropriate name for this function. What about `set_signature_type` ??

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]

File: lib/types/sign_command.ex (hunk 1 above)
Comment (CristhianRodriguezMolina) [reply lib/types/sign_command.ex]: @juanhurtado10 You are right, we'll change it.

File: lib/types/sign_command.ex (hunk 2 above)
Comment (CristhianRodriguezMolina) [reply lib/types/sign_command.ex]: @juanhurtado10 I think it looks better, thank you.

**Review by juanhurtado10 — APPROVED** [review]

**Pull request merged**